### ISSUE TITLE
bash train.sh时报错

### ISSUE THREAD

**Issue by z1968357787** [THUDM/ChatGLM-6B#448]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

运行bash train.sh出现了这个错误

train.sh: line 3: $'\r': command not found
usage: main.py [-h] --model_name_or_path MODEL_NAME_OR_PATH [--config_name CONFIG_NAME] [--tokenizer_name TOKENIZER_NAME] [--cache_dir CACHE_DIR]
               [--use_fast_tokenizer [USE_FAST_TOKENIZER]] [--no_use_fast_tokenizer] [--model_revision MODEL_REVISION]
               [--use_auth_token [USE_AUTH_TOKEN]] [--resize_position_embeddings RESIZE_POSITION_EMBEDDINGS] [--quantization_bit QUANTIZATION_BIT]
               [--pre_seq_len PRE_SEQ_LEN] [--prefix_projection [PREFIX_PROJECTION]] [--lang LANG] [--dataset_name DATASET_NAME]
               [--dataset_config_name DATASET_CONFIG_NAME] [--prompt_column PROMPT_COLUMN] [--response_column RESPONSE_COLUMN]
               [--train_file TRAIN_FILE] [--validation_file VALIDATION_FILE] [--test_file TEST_FILE] [--overwrite_cache [OVERWRITE_CACHE]]
               [--preprocessing_num_workers PREPROCESSING_NUM_WORKERS] [--max_source_length MAX_SOURCE_LENGTH]
               [--max_target_length MAX_TARGET_LENGTH] [--val_max_target_length VAL_MAX_TARGET_LENGTH] [--pad_to_max_length [PAD_TO_MAX_LENGTH]]
               [--max_train_samples MAX_TRAIN_SAMPLES] [--max_eval_samples MAX_EVAL_SAMPLES] [--max_predict_samples MAX_PREDICT_SAMPLES]
               [--num_beams NUM_BEAMS] [--ignore_pad_token_for_loss [IGNORE_PAD_TOKEN_FOR_LOSS]] [--no_ignore_pad_token_for_loss]
               [--source_prefix SOURCE_PREFIX] [--forced_bos_token FORCED_BOS_TOKEN] --output_dir OUTPUT_DIR
               [--overwrite_output_dir [OVERWRITE_OUTPUT_DIR]] [--do_train [DO_TRAIN]] [--do_eval [DO_EVAL]] [--do_predict [DO_PREDICT]]
               [--evaluation_strategy {no,steps,epoch}] [--prediction_loss_only [PREDICTION_LOSS_ONLY]]
               [--per_device_train_batch_size PER_DEVICE_TRAIN_BATCH_SIZE] [--per_device_eval_batch_size PER_DEVICE_EVAL_BATCH_SIZE]
               [--per_gpu_train_batch_size PER_GPU_TRAIN_BATCH_SIZE] [--per_gpu_eval_batch_size PER_GPU_EVAL_BATCH_SIZE]
               [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS] [--eval_accumulation_steps EVAL_ACCUMULATION_STEPS]
               [--eval_delay EVAL_DELAY] [--learning_rate LEARNING_RATE] [--weight_decay WEIGHT_DECAY] [--adam_beta1 ADAM_BETA1]
               [--adam_beta2 ADAM_BETA2] [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--num_train_epochs NUM_TRAIN_EPOCHS]
               [--max_steps MAX_STEPS]
               [--lr_scheduler_type {linear,cosine,cosine_with_restarts,polynomial,constant,constant_with_warmup,inverse_sqrt}]
               [--warmup_ratio WARMUP_RATIO] [--warmup_steps WARMUP_STEPS] [--log_level {debug,info,warning,error,critical,passive}]
               [--log_level_replica {debug,info,warning,error,critical,passive}] [--log_on_each_node [LOG_ON_EACH_NODE]] [--no_log_on_each_node]
               [--logging_dir LOGGING_DIR] [--logging_strategy {no,steps,epoch}] [--logging_first_step [LOGGING_FIRST_STEP]]
               [--logging_steps LOGGING_STEPS] [--logging_nan_inf_filter [LOGGING_NAN_INF_FILTER]] [--no_logging_nan_inf_filter]
               [--save_strategy {no,steps,epoch}] [--save_steps SAVE_STEPS] [--save_total_limit SAVE_TOTAL_LIMIT]
               [--save_on_each_node [SAVE_ON_EACH_NODE]] [--no_cuda [NO_CUDA]] [--use_mps_device [USE_MPS_DEVICE]] [--seed SEED]
               [--data_seed DATA_SEED] [--jit_mode_eval [JIT_MODE_EVAL]] [--use_ipex [USE_IPEX]] [--bf16 [BF16]] [--fp16 [FP16]]
               [--fp16_opt_level FP16_OPT_LEVEL] [--half_precision_backend {auto,cuda_amp,apex,cpu_amp}] [--bf16_full_eval [BF16_FULL_EVAL]]
               [--fp16_full_eval [FP16_FULL_EVAL]] [--tf32 TF32] [--local_rank LOCAL_RANK] [--xpu_backend {mpi,ccl,gloo}]
               [--tpu_num_cores TPU_NUM_CORES] [--tpu_metrics_debug [TPU_METRICS_DEBUG]] [--debug DEBUG]
               [--dataloader_drop_last [DATALOADER_DROP_LAST]] [--eval_steps EVAL_STEPS] [--dataloader_num_workers DATALOADER_NUM_WORKERS]
               [--past_index PAST_INDEX] [--run_name RUN_NAME] [--disable_tqdm DISABLE_TQDM] [--remove_unused_columns [REMOVE_UNUSED_COLUMNS]]
               [--no_remove_unused_columns] [--label_names LABEL_NAMES [LABEL_NAMES ...]] [--load_best_model_at_end [LOAD_BEST_MODEL_AT_END]]
               [--metric_for_best_model METRIC_FOR_BEST_MODEL] [--greater_is_better GREATER_IS_BETTER] [--ignore_data_skip [IGNORE_DATA_SKIP]]
               [--sharded_ddp SHARDED_DDP] [--fsdp FSDP] [--fsdp_min_num_params FSDP_MIN_NUM_PARAMS] [--fsdp_config FSDP_CONFIG]
               [--fsdp_transformer_layer_cls_to_wrap FSDP_TRANSFORMER_LAYER_CLS_TO_WRAP] [--deepspeed DEEPSPEED]
               [--label_smoothing_factor LABEL_SMOOTHING_FACTOR]
               [--optim {adamw_hf,adamw_torch,adamw_torch_fused,adamw_torch_xla,adamw_apex_fused,adafactor,adamw_bnb_8bit,adamw_anyprecision,sgd,adagrad}]
               [--optim_args OPTIM_ARGS] [--adafactor [ADAFACTOR]] [--group_by_length [GROUP_BY_LENGTH]] [--length_column_name LENGTH_COLUMN_NAME]
               [--report_to REPORT_TO [REPORT_TO ...]] [--ddp_find_unused_parameters DDP_FIND_UNUSED_PARAMETERS]
               [--ddp_bucket_cap_mb DDP_BUCKET_CAP_MB] [--dataloader_pin_memory [DATALOADER_PIN_MEMORY]] [--no_dataloader_pin_memory]
               [--skip_memory_metrics [SKIP_MEMORY_METRICS]] [--no_skip_memory_metrics] [--use_legacy_prediction_loop [USE_LEGACY_PREDICTION_LOOP]]
               [--push_to_hub [PUSH_TO_HUB]] [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--hub_model_id HUB_MODEL_ID]
               [--hub_strategy {end,every_save,checkpoint,all_checkpoints}] [--hub_token HUB_TOKEN] [--hub_private_repo [HUB_PRIVATE_REPO]]
               [--gradient_checkpointing [GRADIENT_CHECKPOINTING]] [--include_inputs_for_metrics [INCLUDE_INPUTS_FOR_METRICS]]
               [--fp16_backend {auto,cuda_amp,apex,cpu_amp}] [--push_to_hub_model_id PUSH_TO_HUB_MODEL_ID]
               [--push_to_hub_organization PUSH_TO_HUB_ORGANIZATION] [--push_to_hub_token PUSH_TO_HUB_TOKEN] [--mp_parameters MP_PARAMETERS]
               [--auto_find_batch_size [AUTO_FIND_BATCH_SIZE]] [--full_determinism [FULL_DETERMINISM]] [--torchdynamo TORCHDYNAMO]
               [--ray_scope RAY_SCOPE] [--ddp_timeout DDP_TIMEOUT] [--torch_compile [TORCH_COMPILE]] [--torch_compile_backend TORCH_COMPILE_BACKEND]
               [--torch_compile_mode TORCH_COMPILE_MODE] [--sortish_sampler [SORTISH_SAMPLER]] [--predict_with_generate [PREDICT_WITH_GENERATE]]
               [--generation_max_length GENERATION_MAX_LENGTH] [--generation_num_beams GENERATION_NUM_BEAMS]
main.py: error: the following arguments are required: --model_name_or_path, --output_dir
train.sh: line 5: --do_train: command not found
train.sh: line 6: --train_file: command not found
train.sh: line 7: --validation_file: command not found
train.sh: line 8: --prompt_column: command not found
train.sh: line 9: --response_column: command not found
train.sh: line 10: --overwrite_cache: command not found
train.sh: line 11: --model_name_or_path: command not found
train.sh: line 12: --output_dir: command not found
train.sh: line 13: --overwrite_output_dir: command not found
train.sh: line 14: --max_source_length: command not found
train.sh: line 15: --max_target_length: command not found
train.sh: line 16: --per_device_train_batch_size: command not found
train.sh: line 17: --per_device_eval_batch_size: command not found
train.sh: line 18: --gradient_accumulation_steps: command not found
train.sh: line 19: --predict_with_generate: command not found
train.sh: line 20: --max_steps: command not found
train.sh: line 21: --logging_steps: command not found
train.sh: line 22: --save_steps: command not found
train.sh: line 23: --learning_rate: command not found
train.sh: line 24: --pre_seq_len: command not found
train.sh: line 25: --quantization_bit: command not found

### Expected Behavior

no

### Steps To Reproduce

bash train.sh

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
全都是按requirements里面的要求来的
```

### Anything else?

no

**Comment by duzx16** [comment]
报错信息为`train.sh: line 3: $'\r': command not found`
仓库里的`train.sh`的第三行没有这个内容

**Comment by z1968357787** [comment]
那是什么问题呢，我用的是最新的train.sh，第三行没内容但是就是报错了，说指令没有此命令，跟操作系统版本有关吗？



&nbsp;


变质的水果糖
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhengxiao ***@***.***&gt;; 
发送时间: 2023年4月7日(星期五) 晚上6:52
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [THUDM/ChatGLM-6B] bash train.sh时报错 (Issue #448)





 
报错信息为train.sh: line 3: $'\r': command not found
 仓库里的train.sh的第三行没有这个内容

**Comment by littlerookie** [comment]
文件分隔符问题，把文件分隔符由CRLF改为LF
